linera-execution/src/resources.rs

```diff
@@ -330,6 +330,7 @@ where
         match self.owner {
             None => self.view.system.balance.get_mut().try_add_assign(other),
             Some(owner) => {
+                // Try to credit the owner account first.
```

Comment:
This function is only called in `merge_balance` at the moment, is it?

What if an operation sends a token to the chain account, but also consumes fuel. Won't these two balance changes get merged, and the token then credited to the owner account instead?

Reply:
System API calls will modify the "real" balances in the `SystemExecutionStateView` through the actor messages, not the temporary copy used for resource control.

Reply:
What you describe could happen, if we activate "storage refunds".